build.sbt

```diff
@@ -34,6 +29,31 @@ ThisBuild / wildcardImport := {
   }
 }
 
+// Scalafix
+
+// Use the scalafix config which is valid for all versions of Scala if
+// scalaVersion is different than the default one. Otherwise, use the full
+// scalafix config, which includes Scala version specific rules.
+ThisBuild / scalafixConfig := {
+  if (scalaVersion.value != DefaultScalaVersion) {
+    Some(file(".scalafix-base.conf"))
+  } else scalafixConfig.value
+}
+ThisBuild / scalafixScalaBinaryVersion := {
+  if (scalaVersion.value != DefaultScalaVersion) {
```

Comment:
The unqualified `scalaVersion` here resolves to `ThisBuild / scalaVersion` which is unaffected by `++` IIRC. I think I've fallen into this trap before 😅 using the `LocalRootProject / scalaVersion` is a safe bet.

Reply:
Updated